2022-11-23_proposal-process.md

```diff
@@ -0,0 +1,131 @@
+
+Once PR get merged, the design doc can change, but it requires (less strict, but still) a PR with review and merge by a maintainer.
+
+Two features are present in the current index page: Implementation Status and TODO design docs.
+
+### TODO Proposals
+
+For `TODO's, so ideas for design docs we know we want (e.g. it was decided on the DevSummit). I propose to use GitHub Issues in the `github.com/prometheus/proposals`repository for those with appropriate`TODO` labels.
+
+### Implementation Status
+
+For Implementation Status, I propose a "best effort" `Implementation Status` field in markdown and a roughly maintained list of links to relevant PRs and Issues. We can iterate over it, but without automation, I don't expect owners to always update this field with new changes.
+
+## Alternatives
+
+1. Different placement of design docs in markdown: `prometheus/prometheus`.
+
+We could place them in [Prometheus repo](https://github.com/prometheus/prometheus), bringing more visibility. There would be some issues, though:
+
+* CI checks would get run on every design doc change, which will bring a lot of pain. We could invest in special CI rules to avoid this, but it's not trivial.
+* Design docs buried somewhere in the Prometheus repo will be less discoverable. It also makes it harder to categorize GitHub Discussions and Issue related for design docs.
+* Design docs do not only relate to Prometheus repo but full ecosystem or even neighbour projects like Alertmanager and clients. Ideally, we can share the same design process across all the things Prometheus Team and community help with.
+
+2. Different placement of design docs in markdown: `prometheus/docs`
+
+We could place them in [Prometheus docs](https://github.com/prometheus/docs), which already hosts [index of design docs](https://github.com/prometheus/docs/blob/main/content/docs/introduction/design-doc.md). There are some issues too:
+
+* Still, some CI checks for a website would be unrelated.
+* Design docs buried somewhere in website implementation and content, despite not rendering the proposals on the website. Less discoverable, and it also makes it harder to categorize GitHub Discussions and Issue related for design docs.
+
+3. Stick to Google Docs for design docs.
+
+It's worth exploring the idea of sticking to Google Docs as we do right now and trying to mitigate the flaws mentioned in [Pitfalls of the current solution](#pitfalls-of-the-current-solution) in some way.
+
+* We could ask to use a consistent template.
+* We could maintain Prometheus Google Drive and ask owners to transfer ownership to us, which would give some immutability. Still tracking consensus and maintaining this is non-trivial. Discovery is also poor and would mean extra work to maintain the index page like https://prometheus.io/docs/introduction/design-doc/
+
+4. Have `Accepted`, `Rejected`, and `Implemented` directories for different statuses
```

Comment:
I like this a lot. If we take a quick look at https://prometheus.io/docs/introduction/design-doc/ we have `Partially implemented`, `Approved`, `Implemented`, `Under review`, `Proposed`, `In progress`, `TODO`.

I tried to keep this updated while working out-of-order and it's true that I almost forgot about it.

I wonder if showing what's being worked on would add some value. It's one thing to have a proposal accepted i.e some proposal discussed with the prometheus team versus someone actively working on that proposal.

Maybe we need a 4th directory/status `In Progress` ? For example `Native Histograms` would be in progress today.

Reply:
Yeah I like that too, having a status tracking the progress in the proposal itself :+1:  My bad that I did not get the format on the first go.